guardian/pinboard
# add/configure CDK to produce CloudFormation

## What does this change?
This adds a new directory `cdk`, containing everything necessary to build our CloudFormation from CDK code. So far the infrastructure items are...

- Lamba (the `bootstrapping-lambda` - see #1 )
- API Gateway (for the above Lambda)
- AppSync GraphQL API configuration (which loads a `schema.graphql` file containing the current schema)
- DynamoDB table (which sits behind the above AppSync API)

## How to test
In the `cdk` directory, run `yarn` followed by `yarn synth` and notice the generated `cloudformation.yaml` file.

## How can we measure success?
He we can have codified infrastructure which is type checked (and even unit testable, should we wish in future).

## Have we considered potential risks?
Until subsequent PRs this won't be executed and so poses no risks at this time.

## Images
N/A
